### PR TITLE
Hoist OperationStack, OperationError and utils from Solution.js

### PR DESCRIPTION
**Pull request by dbouwman**
Adds:
- OperationStack & OperationError
- Operation Stack Doc
- last, uniqueBy, filterBy

Pulled from new Solution.js, but need here for use in sites package

### CODE REVIEW

File: packages/common/test/util.test.ts [hunk 1]
```diff
@@ -675,4 +678,40 @@ describe("util functions", () => {
...
+      const apps = filterBy(arr, "item.type", "App");
+      expect(apps.length).toBe(2, "should have two apps");
+      apps.forEach(app => {
+        expect(app.item.type).toBe("App");
+      });
+    });
+  });
+
+  describe("uniqueBy: ", () => {
+    it("returns uniqu entries by deep prop", () => {
```
Comment (tomwayson) [inline packages/common/test/util.test.ts:706]:
```suggestion
    it("returns unique entries by deep prop", () => {
```

File: packages/common/test/util.test.ts [hunk 2]
```diff
@@ -675,4 +678,40 @@ describe("util functions", () => {
...
+      const arr = [
+        { item: { type: "Map", id: "map1" } },
+        { item: { type: "App", id: "app1" } },
+        { item: { type: "Map", id: "map1" } },
+        { item: { type: "App", id: "app2" } }
+      ];
+      const chk = uniqueBy(arr, "item.id");
```
Comment (tomwayson) [inline packages/common/test/util.test.ts:713]: seems like `uniqueBy(arr, "item.type")` would be a better example just b/c we expect id to be unique across all items.

**Review by tomwayson — APPROVED** [review]
Good stuff. Stack docs are a help. Left minor comments that may not be worth addressing in this PR.

File: packages/common/test/util.test.ts (hunk 2 above)
Comment (dbouwman) [reply packages/common/test/util.test.ts:713]: I'll mull this - how I use it, and why I wrote it, was actually to unique a set of `IDependency` objects, by Item id, so we don't try to convert the same item multiple times... but yeah - in general we'd not expect to have dupes of item-ish things

**Pull request merged**